opentelekomcloud/resource_opentelekomcloud_dds_instance_v3.go

```diff
@@ -0,0 +1,561 @@
+package opentelekomcloud
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/opentelekomcloud/gophertelekomcloud"
+	"github.com/opentelekomcloud/gophertelekomcloud/openstack/dds/v3/instances"
+)
+
+func resourceDdsInstanceV3() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceDdsInstanceV3Create,
+		Read:   resourceDdsInstanceV3Read,
+		Update: resourceDdsInstanceV3Update,
+		Delete: resourceDdsInstanceV3Delete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"datastore": {
+				Type:     schema.TypeList,
+				Required: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"DDS-Community",
+							}, true),
+						},
+						"version": {
+							Type:     schema.TypeString,
+							Required: true,
+							ForceNew: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"3.4", "3.2",
```

Comment:
It doesn't seem to be a good idea to validate version range, as version tend to change between OTC releases